### ISSUE TITLE
Feature: Ability to manually specify quantity for shopping list

### ISSUE THREAD

**Issue by rosscarlson** [msarilar/EDEngineer#138]
Rather than having to press + over and over for large items it would be nice to be able to just type in a number.  For example on the lower level blueprints you might want to be able to roll something 25-30 times to get to higher levels with that engineer.  Picking a simple level 1 item and doing it 25+ times works well so I have to hit + 25 times right now rather than just typing in 25.

Minor but would be nice.

**Comment by msarilar** [comment]
when I thought about the feature, I did not expect users adding 25 times a blueprint in their shopping list ; why would you do such thing ? 🤔

**Comment by rosscarlson** [comment]
Oh I'm just stupid and like pain ;-)  Now to explain.

I did some missions at 17 Draconis to make a bunch of money.  I realized that I was getting paid in a bunch of somewhat hard to get Engineering commodities.  So I decided I'd do some engineering.  One thing led to another and as I've done very little so far I had 5 Engineers unlocked with a lot of various blueprints very close to done.  So I kept collecting and figured out each item I wanted at each Engineer.  That wound up being about 20 different blueprints and in a bunch of cases I'd want 3-5 of each type (say 4 FSDs, 3 lasers, etc).  That way I could do get a class 3, 4 ,5 and 6 FSD and put them on the shelf to share between ships.  I'm now 3 materials from being done and need just a few more of 3 others to do them all.  The beauty of the shopping list not only tells me I need X item but that if I'm going to do all this stuff I need of X item.  Brilliant.

Okay, so then I realize that a few of the Engineers I'll need to push to level 5 so I'll need to do a bunch of level 1 and maybe 3 rolls to do that.  So I figure I'll need to do say 25 level 1, or at least I want to be able to.  So I want to do 25 rolls using Nickle, 25 using Sulfur, 25 Tin, etc. etc.  So I added the corresponding grade 1 blueprint 25 times. 

Make sense?  Obviously this is probably not the usual way it's done so I wouldn't complain in the slightest if this was closed with a "you're crazy, just click + 25 times if you're that stupid".  I suppose I hoped that the number field there was easy enough to make "typable" like the fields are when entering item from your ships inventory the first time.  If it's any harder than that even I'd say close this.

I think I mentioned I'm rather detailed with this kinda stuff and I'm also, uh, known for being a little extreme.  I'm a rather good tester as I tend to think of things no one else would be likely to try...
